tensorflow_addons/rnn/cell.py

```diff
@@ -366,3 +366,425 @@ def _create_norm_layer(self, name):
+            or
+              net = x_t * W_xh + y_{t-1} * W_hh
+              y_t = f(net + b)
+
+        Case 2:
+            RNN with, layer normalization (only scaling), bias and activation.
+              y_t = f(ln(x_t * W_xh + y_{t-1} * W_hh) + b)
+            or
+              net = x_t * W_xh + y_{t-1} * W_hh
+              y_t = f(ln(net) + b)
+
+            Layer normalization with scaling and centering in one go (see Ba et
+            al (2016), page 3, formula 4, https://arxiv.org/abs/1607.06450)
+            is the same as layer normalization only with scaling, and
+            centering directly afterwards.
+
+        Case 3:
+            RNN, with dropout, bias, and activation (no scaling from LN)
+              y_t = f(d1(x_t) * W_xh + d2(y_{t-1}) * W_hh + b)
+            or
+              net = d1(x_t) * W_xh + d2(y_{t-1}) * W_hh
+              y_t = f(net + b)
+
+        Case 4:
+            Everyting is used, i.e. all dropouts, layer normalization
+            (only scaling), bias, and activation
+              y_t = f(ln(d1(x_t) * W_xh + d2(y_{t-1}) * W_hh) + b)
+            or
+              net = d1(x_t) * W_xh + d2(y_{t-1}) * W_hh
+              y_t = f(ln(net) + b)
+        """
+        prev_output = states[0]
+        dp_mask = self.get_dropout_mask_for_cell(inputs, training)
+        rec_dp_mask = self.get_recurrent_dropout_mask_for_cell(
+            prev_output, training)
+
+        if dp_mask is not None:
+            h = keras.backend.dot(inputs * dp_mask, self.kernel)
+        else:
+            h = keras.backend.dot(inputs, self.kernel)
+
+        # don't add bias to "h" here
+        # add bias after scaling with layer normalization to "output"
+
+        if rec_dp_mask is not None:
+            prev_output = prev_output * rec_dp_mask
+        output = h + keras.backend.dot(prev_output,
+                                       self.recurrent_kernel)  # "net"
+
+        output = self.layernorm(output)
+
+        if self.bias is not None:
+            output = keras.backend.bias_add(output, self.bias)
+
+        if self.activation is not None:
+            output = self.activation(output)
+
+        return output, [output]
+
+    # use SimpleRNNCell's get_initial_state method
+
+    def get_config(self):
+        cell_config = super(LayerNormSimpleRNNCell, self).get_config()
+        del cell_config['name']
+
+        ln_config = self.layernorm.get_config()
+        ln_config = {
+                key: ln_config[key]
+                for key in [
+                    "epsilon", "gamma_initializer", "gamma_regularizer",
+                    "gamma_constraint"
+                ] if key in ln_config
+        }
+        ln_config['layernorm_epsilon'] = ln_config.pop("epsilon")
+        return dict(list(cell_config.items()) + list(ln_config.items()))
+
+
+@tf.keras.utils.register_keras_serializable(package='Addons')
+class LayerNormSimpleRNN(keras.layers.SimpleRNN):
```

Comment:
Thinking about this more, I am not so sure whether this layer is needed or not, since it can be simply replaced by RNN(LayernormSimpleRNNCell(xxx)).

Reply:
agree. The cell's docstring has an example how to use the cell class. I am going to delete LayerNormSimpleRNN